.github/workflows/main.yml

```diff
@@ -0,0 +1,66 @@
+name: Pull Request
+
+on:
+  pull_request:
+    types: [opened, synchronize]
+
+env:
+  NODE_VERSION: 21.5.0
+
+jobs:
```

Comment:
To me it seems suboptimal to create three separate jobs for each action and run checkout, install, etc. each time

Reply:
I'd add a build step

Reply:
Could you, please, specify more in depth what do you mean by build step. Did you mean to cache npm dependencies?

or to compose everything into one big step? 
```
  build:
    name: Build
    runs-on: ubuntu-latest

    steps:
      - name: Checkout
        uses: actions/checkout@v3

      - name: Setup Node.js
        uses: actions/setup-node@v3
        with:
          node-version: ${{ env.NODE_VERSION }}

      - name: Install dependencies
        run: npm install

      - name: Lint Application
        run: npm run lint

      - name: Check Application Types
        run: npm run type-check

      - name: Run application tests
        run: npm run test
```

Reply:
I just meant that you can add:
```
      - name: Build
        run: npm run build
```
after installing deps

Reply:
I don't get it, why do we want to build the app

Reply:
Just to check that it can be compiled without any problems, because I think `tsc --noEmit` might not catch all possible problems. 
If you decide to make the proposed change, I think the Check Application Types step can be removed

Reply:
Just to clarify - we want to switch the "Check Application Types" step with proposed "Build" step, but linting and test steps leave as it was before?

Reply:
Got it, thank you!

Reply:
I think there's still some misunderstanding between us 🙂. I meant that there should be only one job and it should also contain a build step, that's why I left two separate comments

Reply:
In this case we are gonna left with only one displayed block of information that has passed the check and that has not, without separation. Are we fine with that?
![image](https://github.com/rolling-scopes/site/assets/114234698/17d1031b-33c8-4664-86ce-01f1c38bfcce)

Reply:
But since the PR is already merged, it's ok